added animal index page and animal MVC

mcat115 opened the pull request:
Added index to routes
Made animal model (validates name, body, and rating)
Made animal migration  (animal has rating (which will be aggregate rating), body (for description of animal), and name)
Added index to routes
Made view for index
Made react router
Made index component

dovakeen118 commented on app/controllers/animals_controller.rb:
```diff
@@ -0,0 +1,7 @@
+class AnimalsController < ApplicationController
```
This is not an API controller? Take a look at my other comments below where you have both a React component for the index and an ERB template. From this controller it looks like you intend to use the ERB template, in which case you can delete the component, or vice versa. In either case, your view (template or component) should be set up to display the animals data

dovakeen118 commented on app/javascript/react/components/AnimalIndex.js:
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+const AnimalIndex = (props) => {
+  return (
+    <h1>Animals</h1>
```
No animals being displayed yet? For this feature of an animals index page, we wouldn't consider it complete if we aren't able to see that data that you set up on the backend!

dovakeen118 commented on db/migrate/20210421141639_create_animals.rb:
```diff
@@ -0,0 +1,9 @@
+    create_table :animals do |t|
+      t.string :name, null: false
+      t.text :body, null: false
+      t.float :rating, default: 0, null: false
```
TIMESTAMPS!!

mcat115 replied:
I added timestamps last minute but didn't commit... smh

dovakeen118 commented on app/views/animals/index.html.erb:
```diff
@@ -0,0 +1 @@
+<h1>This will be replaced by React</h1>
```
You have both an index ERB and an index React component - I would choose one or the other

dovakeen118 reviewed (changes requested):
Looks like a little more to be added for this feature - also make sure your migrations have timestamps!

mcat115 commented:
Since the last pull request, we have:
Deleted the erbs what were not necessary because they were being handled by react
Committed again with timestamps in the commit this time
Added ApiController
Added a fetch request and useEffect to animal index.js
Added a map function to return animal tiles for when we make those
